app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AmendRegistrationStartController.scala

```diff
@@ -61,6 +70,22 @@ class AmendRegistrationStartController @Inject() (
       }
   }
 
+  private def routeToAmendReason(eclRegistrationReference: String, internalId: String)(implicit
+    hc: HeaderCarrier
+  ): Future[Result] = {
+    val result = for {
+      registration            <- registrationService.getOrCreateRegistration(internalId)
+      getSubscriptionResponse <- registrationService.getSubscription(eclRegistrationReference)
+      upsertedRegistration    <-
+        registrationService.upsertRegistration(
+          registrationService.transformToRegistration(registration, getSubscriptionResponse)
```

Comment:
can the result of registrationService.transformToRegistration(registration, getSubscriptionResponse) be set to a val in the for-comprehension? It's easier to process than a call within a call:

```
registration = registrationService.transformToRegistration(registration, getSubscriptionResponse)
_ <-   registrationService.upsertRegistration(registration)
```

Reply:
Done